docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -129,6 +130,30 @@ Msg (<int>/<int>): <string>           // E.g. Msg (1/2): bank v1beta1 send coins
 End of transaction messages
 ```
 
+### 9. Custom `Msg` Renderers
+
+Application developers may choose to not follow default renderer value output for their own `Msg`s. In this case, they can implement their own custom `Msg` renderer. This is similar to [EIP4430](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4430.md), where the smart contract developer chooses the description string to be shown to the end user.
+
+This is done by setting the `cosmos.msg.v1.textual.custom_renderer` Protobuf option to `true`. This option CAN ONLY be set on a Protobuf message representing transaction message object (implementing `sdk.Msg` interface).
```

Comment:
Is the intention to communicate this to clients with the proto option? Is so we should say this and also maybe have a descriptive proto field to explain the format

Reply:
Yes, it's mainly to ease communication between app devs and client devs. So we currently have these ideas:
1. Boolean: `custom_renderer = true`
2. An unique algorithm identifier: `custom_renderer = "agoric_msg_to_text_v1"` (proposed by @webmaster128)
3. Human readable format: `custom_renderer = "Username: <username_from_state>\nValue: <value>"` (if I understand your proposal @aaronc, this string is not meant to be parsed programatically, it's just for humans).

I went with bool initially, but slowly changing my mind towards some kind of human-readable string.

Reply:
Given the concerns about the feature, consider requiring a string like "I know what I'm doing" or "Don't blame Cosmos-SDK". `:-)`

Reply:
I went with 2, but also proposing like Jim said a "I know what I'm doing" warning: https://github.com/cosmos/cosmos-sdk/pull/12785#discussion_r937985861